### DISCUSSION TITLE
Git Bash screen tearing issue

### DISCUSSION THREAD

**Discussion by eltsu7** [git-for-windows/git#4541] [Q&A] [answered]
Hey, I'm noticing git bash has screen tearing when using terminal user interfaces, through SSH to be clear. This does not happen on windows command line or powershell. Same thing happens on windows 10 and 11. Wondering if this is git bash related or the software I'm building this on (could it even be?).

I tried git bash with cygwin and xterm, both had the same problem. In the video below I'm using git bash and powershell, both with xterm.

https://github.com/git-for-windows/git/assets/23433034/e730a575-ee91-46cf-b18d-e096daa1e903

**Comment by dscho** [comment]
I am afraid that I do not understand what you mean by "screen tearing". Care to elaborate?

**Comment by rimrul** [comment, marked as answer]
I guess you're talking about the big black flickering area in the video. If you look closely you can see a similar but smaller artefact around 0:09 on the powershell tab. This might just be down to timings and performance. Could you check wether you're using the same ssh.exe in both tabs?

**Reply by eltsu7** [reply]
Thank you for pointing in the right direction! Found out git bash uses it's own ssh client in ...\AppData\Local\Programs\Git\usr\bin\ssh.exe and others use OpenSSH in C:\Windows\System32\OpenSSH\ssh.exe. I tried launching the latter one from git bash and it worked the way I wanted.

I'm guessing gits ssh client clears the screen and then writes where as openssh just rewrites on top of the old characters.
